src/helpers/Helpers.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Commerce Insights Components Helpers class
+ *
+ * @link      https://fostercommerce.com
+ * @copyright Copyright (c) 2021 Foster Commerce
+ */
+
+declare(strict_types = 1);
+
+namespace fostercommerce\commerceinsights\helpers;
+
+use DateTime;
+use NumberFormatter;
+use Money\Money;
+use Money\Currency;
+use Money\Currencies\ISOCurrencies;
+use Money\Formatter\IntlMoneyFormatter;
+
+use Craft;
+
+class Helpers {
+    public static function getDateRangeData(): array {
+        $result = [];
+
+        // Query params if they exist in the URL
+        $startQuery = Craft::$app->request->getQueryParam('startDate') ?? null;
+        $endQuery   = Craft::$app->request->getQueryParam('endDate') ?? null;
+
+        // URL decode the query params if they are present
+        // TODO: This might override the selected range? Or vice versa. Not implemented yet.
+        $result['startQuery'] = $startQuery ? urldecode($startQuery) : null;
+        $result['endQuery']   = $endQuery ? urldecode($endQuery) : null;
+
+        // All the formatted dates for the selected range (currently selected period)
+        // If none is selected, we just do a week
+        $result['today']      = new DateTime(date('Y-m-d 23:59:59'));
+        $result['weekAgo']    = $result['today']->modify('-7 day')->format('Y-m-d 00:00:00');
+        $result['rangeStart'] = Craft::$app->request->getBodyParam('range_start') ?? null;
+
+        // Either the selected end date, or today
+        $result['rangeEnd'] = Craft::$app->request->getBodyParam('range_end') ?? $result['today']->format('Y-m-d 23:59:59');
```

Comment:
`range_start` and `range_end` are sent via `POST`, and `startDate` and `endDate` are possible URI query parameters.

Reply:
Why differentiate between the 2 HTTP methods for the properties? It looks like there's specific logic for a GET vs for POST, but for GET requests, only startQuery is set but it is not used anywhere else. 

Consider using [`Request::getParam`](https://github.com/craftcms/cms/blob/develop/src/web/Request.php#L1044-L1055) which will look use the query parameter if present, otherwise try the request body.